globals/matchmaking_globals.go

```diff
@@ -52,7 +51,7 @@ func FindSearchMatchmakeSession(searchMatchmakeSession match_making.MatchmakeSes
 	//this portion finds any sessions that match the search session. It does not care about anything beyond that, such as if the match is already full. This is handled below.
 	candidateSessionIndexes := make([]uint32, 0, len(Sessions))
 	for index, session := range Sessions {
-		if reflect.DeepEqual(session.SearchMatchmakeSession, searchMatchmakeSession) { // TODO - for Jon: Equals in StructureInterface
+		if session.SearchMatchmakeSession.Equals(&searchMatchmakeSession) {
```

Comment:
I see a lot of referencing and dereferencing going on all throughout here, is there a reason for that? We usually just stick with passing pointers around, mostly because it's faster with larger data structures (and we have very large and complex data structures at times)

Reply:
The Session globals store the `MatchmakeSession` without pointers, so I adapted the code to what was done.

I will change it to pointers

Reply:
Gotcha, I must not have caught that in a previous commit. That's my bad, thank you